### DISCUSSION TITLE
I need help using ethers.js.

### DISCUSSION THREAD

**Discussion by cleverbotdev** [community/community#63376] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

I sometimes get null when using provider.getTransaction(hash) function in ethers.

`` const tx = await provider.getTransaction(hash); ``

How can I fix this issue?

**Comment by workwithhim** [comment, marked as answer]
I got the same issue when using a pending transaction hash.
I solved this problem by using provider.waitForTransaction(hash)
 `` const tx = await provider.waitForTransaction(hash) ``

**Comment by cleverbotdev** [comment]
> I got the same issue when using pending hash. I solved this problem by using provider.waitForTransaction(hash) `const tx = await provider.waitForTransaction(hash)`
Thanks @workwithhim
